internal/reconciler/telemetry/metric_components_checker.go

```diff
@@ -77,7 +83,7 @@ func (m *metricComponentsChecker) createMessageForReason(pipelines []v1alpha1.Me
 
 func (m *metricComponentsChecker) createConditionFromReason(reason, message string) *metav1.Condition {
 	conditionType := "MetricComponentsHealthy"
-	if reason == conditions.ReasonMetricGatewayDeploymentReady || reason == conditions.ReasonNoPipelineDeployed {
+	if reason == conditions.ReasonMetricGatewayDeploymentReady || reason == conditions.ReasonNoPipelineDeployed || reason == conditions.ReasonMetricAgentDaemonSetReady {
```

Comment:
"Do we really need this? It seems like 'ReasonMetricAgentDaemonSetReady' will never be used, right?

Reply:
`ReasonMetricAgentDaemonSetReady` required when agent is enabled so agent health check will be last condition of metric pipeline